File: April Leetcode Challenge/31. Next Permutation.java
class Solution {
    public void reverse(int[] nums,int i,int j)
    {
        while(i<=j)
        {
            int t=nums[i];
            nums[i++]=nums[j];
            nums[j--]=t;
        }
    }
    public void nextPermutation(int[] nums) {
        int i,j;
        int n=nums.length;
        for(i=n-2;i>=0;i--)
        {
            if(nums[i]<nums[i+1])
            {
                break;
            }
        }
        if(i<0)
        {
           reverse(nums,0,n-1);
        }
        else
        {
            for(j=n-1;j>=0;j--)
            {
                if(nums[j]>nums[i])
                {
                    break;
                }
            }
            int t=nums[i];
            nums[i]=nums[j];
            nums[j]=t;
            reverse(nums,i+1,n-1);
        }
    }
}

File: Striver_SDE_Sheet/31_Next_Permutation.cpp
//https://leetcode.com/problems/next-permutation/

class Solution {
public:
    void nextPermutation(vector<int>& nums) {
     int n = nums.size();
     int index,number;
    for(index=n-2;index>=0;index--)
    {
     
      if(nums[index]<nums[index+1])
      {
          number = nums[index];
          break;
       }
    }
    if(index<0)
    {
        
    reverse(nums.begin(),nums.end());
    }
    else
    {
    int j;
    for(j=n-1;j>=0;j--)
    {
        if(nums[index] <nums[j])
        {
            break;
        }
    }
        //cout<<index<<j<<"\n";
     swap(nums[index],nums[j]);
     sort(nums.begin()+index+1,nums.end());
        }
    }
};

